design/variable_command_design.md

```diff
@@ -0,0 +1,181 @@
+# Dockerless CLI - variable
+
+This document is the design document for the variable CLI command for the Dockerless project. The functionality for the variable CLI command already exists so all that is left is the following:
+
+1. Ensure UTs and integration tests exist and are GA quality
+2. Improve error handling experience
+3. Write proper documentation in `README.md` of the repository
+
+
+
+```bash
+Name
+----
+  variable - Manage variables
+
+
+Usage
+----
+  conjur variable get [arguments...]
+
+  conjur variable set [arguments...]
+
+
+Global options
+----
+  -h, --help show this help message and exit
+
+
+Commands
+----
+  get       Get the value of a variable can be applied to get the value of several variables
+
+  set       Set the value of a variable
+
+To get help on a specific command, type: conjur <command> -h
+```
+
+
+
+### UX mapping
+
+The following is a mapping of the current flow for all possible permutations of the variable CLI command and their respective expected outputs.
+
+*Input for variable is neither get/set*
+
+```bash
+$ conjur-cli variable boo
+
+## Current
+usage: conjur-cli variable [-h] {get,set} ...
+conjur-cli variable: error: argument action: invalid choice: 'boo' (choose from 'get', 'set')
+
+## Should be
+Error: Command 'variable' requires a subcommand get, set
+
+<Variable command help screen>
+```
+
+*One of the batch secrets doesn’t have value/doesn’t exist*
+
+```bash
+$ conjur-cli variable get secrets/test_secret foo/bar
+
+## Current
+requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://localhost/secrets/myorg/variable/secrets%2Ftest_secret%2Cfoot%2Fbar
+
+## Should be
+Error: Data either not found or variable value does not exist
+
+<Variable command help screen>
+```
+
+*Set secrets*
+
+```bash
+$ conjur-cli variable set foo/bar mysecret
+
+## Current
+Value set: 'foo/bar'
+
+## Should be
+Success! Data written to: 'foo/bar'
+```
+
+*Get secrets*
+
+```bash
+$ conjur-cli variable get foo/bar
+mysecret%
+```
+
+*Get a secret without inputting which secret to get*
+
+```bash
+$ conjur-cli variable get __
+
+## Current 
+usage: conjur-cli variable get [-h] variable_id [variable_id ...]
+
+## Should be
+Error: Command 'variable' requires the following arguments: variable_id
```

Comment:
This command accepts one-or-more variable IDs so the description needs to describe that wording. I do think that the original version of the help here is much more helpful for a CLI user.

Reply:
This was a mistake, the current error is:
```
usage: conjur-cli variable get [-h] variable_id [variable_id ...]
conjur-cli variable get: error: the following arguments are required: variable_id
```